db-tests/src/test/java/gov/nasa/jpl/aerie/database/UserSequenceTests.java

```diff
@@ -0,0 +1,286 @@
+      final var sequences = new ArrayList<UserSequence>();
+      final var res = statement.executeQuery(
+          //language=sql
+          """
+          SELECT id, name, definition, parcel_id, is_locked, workspace_id, owner
+          FROM sequencing.user_sequence
+          WHERE id = %d
+          """.formatted(sequenceId)
+      );
+
+      while (res.next()) {
+        sequences.add(new UserSequence(
+            res.getInt("id"),
+            res.getString("name"),
+            res.getString("definition"),
+            res.getInt("parcel_id"),
+            res.getBoolean("is_locked"),
+            res.getInt("workspace_id"),
+            res.getString("owner")));
+      }
+      return sequences;
+    }
+  }
+  // endregion
+
+  //region Records
+  private record UserSequence(int id, String name, String definition, int parcel_id, boolean is_locked, int workspace_id, String owner) {}
+  //endregion
+
+  @Test
+  void addUserSequence() throws SQLException {
+    final var userSequence = getUserSequences(sequenceId);
+
+    final var expectedUserSequence = new ArrayList<UserSequence>(1);
+    expectedUserSequence.add(new UserSequence(sequenceId, "Test Sequence", sequenceDefinition, parcelId, false, workspaceId, sequenceUser.name()));
+
+    assertEquals(expectedUserSequence, userSequence);
+  }
+
+  @Test
+  void updateUserSequenceParcel() throws SQLException {
+    final var testParcelId = createParcel("Test Parcel 2");
+    assertDoesNotThrow(()->assignParcelToSequence(testParcelId, sequenceId));
+
+    assignParcelToSequence(parcelId, sequenceId);
+  }
+
+  @Test
+  void lockUserSequence() throws SQLException {
+    updateUserSequenceIsLocked(sequenceId, true);
+
+    try {
+      updateUserSequenceName(sequenceId, "Foo Sequence");
+    } catch (SQLException ex) {
+      if (!ex.getMessage().contains("Cannot update locked user sequence.")) {
+        throw ex;
+      }
+    }
+  }
+
+  @Test
+  void lockDoesNotAffectOtherSequence() throws SQLException {
+    final var otherSequenceId = createUserSequence("Test Sequence 2", parcelId, workspaceId, sequenceDefinition, sequenceUser.name());
+
+    assertDoesNotThrow(()->updateUserSequenceName(otherSequenceId, "Bar Sequence"));
+  }
+
+  @Test
+  void unlockUserSequence() throws SQLException {
+    updateUserSequenceIsLocked(sequenceId, false);
+
+    assertDoesNotThrow(()->updateUserSequenceName(sequenceId, "Foo Sequence"));
+
+    final var userSequence = getUserSequences(sequenceId);
+    assertEquals(false, userSequence.get(0).is_locked);
+  }
+
+  @Test
+  void cannotDeleteLockedSequence() throws SQLException {
+    updateUserSequenceIsLocked(sequenceId, true);
+
+    try {
+      deleteUserSequence(sequenceId);
+    } catch (SQLException ex) {
+      if (!ex.getMessage().contains("Cannot delete locked user sequence.")) {
+        throw ex;
+      }
+    }
+  }
+
+  @Test
+  void canDeleteLockedSequence() throws SQLException {
```

Comment:
Minor: name is incorrect, should be:
```suggestion
  void canDeleteUnlockedSequence() throws SQLException {
```